crates/ruff/src/resolve.rs

```diff
@@ -58,11 +63,9 @@ pub fn resolve(
     // that directory. (With `Strategy::Hierarchical`, we'll end up finding
     // the "closest" `pyproject.toml` file for every Python file later on,
     // so these act as the "default" settings.)
-    if let Some(pyproject) = pyproject::find_settings_toml(
-        stdin_filename
-            .as_ref()
-            .unwrap_or(&path_dedot::CWD.as_path()),
-    )? {
+    if let Some(pyproject) =
+        pyproject::find_settings_toml(stdin_filename.as_ref().unwrap_or(&cwd.as_ref()))?
```

Comment:
nit: You do need `as_ref` currently, but `stdin_filename` is already an `Option<&Path>`, so you can actually simplify both of these cases (here and below on line 137) to this:

```suggestion
        pyproject::find_settings_toml(stdin_filename.unwrap_or(&cwd))?
```